lib/Resque.php

```diff
@@ -11,196 +11,330 @@
+                'queue' => $queue,
+            ]);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Reserve and return the next available job in the specified queue.
+     *
+     * @param string $queue Queue to fetch next available job from.
+     * @return Resque_Job Instance of Resque_Job to be processed, false if none or error.
+     */
+    public static function reserve($queue)
+    {
+        require_once dirname(__FILE__) . '/Resque/Job.php';
+
+        return Resque_Job::reserve($queue);
+    }
+
+    /**
+     * Get an array of all known queues.
+     *
+     * @return array Array of queues.
+     */
+    public static function queues()
+    {
+        $queues = self::redis()->smembers('queues');
+        if (!is_array($queues)) {
+            $queues = [];
+        }
+
+        return $queues;
+    }
+
+    /**
+     * Remove Items from the queue
+     * Safely moving each item to a temporary queue before processing it
+     * If the Job matches, counts otherwise puts it in a requeue_queue
+     * which at the end eventually be copied back into the original queue
+     *
+     * @private
+     *
+     * @param string $queue The name of the queue
+     * @param array $items
+     * @return integer number of deleted items
+     */
+    private static function removeItems($queue, $items = [])
+    {
+        $counter = 0;
+        $originalQueue = 'queue:' . $queue;
+        $tempQueue = $originalQueue . ':temp:' . time();
+        $requeueQueue = $tempQueue . ':requeue';
+
+        // move each item from original queue to temp queue and process it
+        $finished = false;
+        while (!$finished) {
+            $string = self::redis()->rpoplpush($originalQueue, self::redis()->getPrefix() . $tempQueue);
+
+            if (!empty($string)) {
+                if (self::matchItem($string, $items)) {
+                    self::redis()->rpop($tempQueue);
+                    $counter++;
+                } else {
+                    self::redis()->rpoplpush($tempQueue, self::redis()->getPrefix() . $requeueQueue);
+                }
+            } else {
+                $finished = true;
+            }
+        }
+
+        // move back from temp queue to original queue
+        $finished = false;
+        while (!$finished) {
+            $string = self::redis()->rpoplpush($requeueQueue, self::redis()->getPrefix() . $originalQueue);
+            if (empty($string)) {
+                $finished = true;
+            }
+        }
+
+        // remove temp queue and requeue queue
+        self::redis()->del($requeueQueue);
+        self::redis()->del($tempQueue);
+
+        return $counter;
+    }
+
+    /**
+     * matching item
+     * item can be ['class'] or ['class' => 'id'] or ['class' => {:foo => 1, :bar => 2}]
+     * @private
+     *
+     * @params string $string redis result in json
+     * @params $items
+     *
+     * @return (bool)
+     */
+    private static function matchItem($string, $items)
```

Comment:
What does `$items` exactly represent ? How is it used ?

I got the impression that it is not the same args as the one passed in `push()`.